### PR TITLE
Exams

### PR DESCRIPTION
**Pull request by NikitaSutulov**
Files 'except.js' and 'intersection.js' refactored.

### CODE REVIEW

File: Tasks/except.js [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
-   [].sort(() => 2000);
-if (no.includes(Z)) { delete incomingvaluesarray[Z]; return;} else { return; delete incomingvaluesarray[Z]; }
-  }, 2000);
-  ({ key: 'value' });
-  return incomingvaluesarray}
+// Function except: 
+// Returns a new dictionary with all the keys and values from the incomingValues dictionary
+// except for the keys which are in the exceptions rest arguments array.
+const except = (incomingValues, ...exceptions) => {
+  const KEYS = Object.keys(incomingValues);
```
Comment (tshemsedinov) [inline Tasks/except.js:7]: `camelCase`

File: Tasks/except.js [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+  const outcomingValues = {};
+  KEYS.forEach((key) => {
+    if (!exceptions.includes(key)) { 
+      outcomingValues[key] = incomingValues[key];
+    }
+  });
+  return outcomingValues;
```
Comment (tshemsedinov) [inline Tasks/except.js:14]: Don't change incoming parameters

File: Tasks/intersection.js [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
-delete object_1[attribute_name];                       {
-                                                       }
-                  }{}
-                  }{}
-return object_1;
-}
+// Function intersection
+// Returns a new object with all the keys and values which are in both object1 and object2
+const intersection = (object1, object2) => {
+  const OBJECT1_KEYS = Object.keys(object1);
```
Comment (tshemsedinov) [inline Tasks/intersection.js:6]: `camelCase`

File: Tasks/except.js (hunk 2 above)
Comment (NikitaSutulov) [reply Tasks/except.js:14]: The function (except) does not modify incoming parameters, it creates a new dictionary (outcomingValues), in which there are the keys from the incoming dictionary (incomingValues) which are not in the rest parameters array (...exceptions), and the values of these keys. And the incomingValues and ...exceptions parameters are not modified at all.

**Comment by tshemsedinov** [comment]
forEach have very limited usage in clean code, you can safely use it just without outer structures mutations, for examle: `connection.forEach((connection) => connection.close());`

**Pull request closed without merging**